tee-worker/ts-tests/common/utils/assertion.ts

```diff
@@ -0,0 +1,206 @@
+    // check identity in idgraph
+    const expected_identity = api.createType(
+        'LitentryIdentity',
+        await buildIdentityHelper(u8aToHex(signer.addressRaw), 'LitentryRococo', 'Substrate')
+    ) as LitentryIdentity;
+    assert.isTrue(isEqual(event.idGraph[0][0], expected_identity));
+    // check identityContext in idgraph
+    assert.equal(event.idGraph[0][1].linking_request_block, 0);
+    assert.equal(event.idGraph[0][1].verification_request_block, 0);
+    assert.isTrue(event.idGraph[0][1].is_verified);
+}
+
+export function assertIdentityVerified(signer: KeyringPair, eventDatas: IdentityGenericEvent[]) {
+    let event_identities: LitentryIdentity[] = [];
+    let idgraph_identities: LitentryIdentity[] = [];
+    for (let index = 0; index < eventDatas.length; index++) {
+        event_identities.push(eventDatas[index].identity);
+    }
+    for (let i = 0; i < eventDatas[eventDatas.length - 1].idGraph.length; i++) {
+        idgraph_identities.push(eventDatas[eventDatas.length - 1].idGraph[i][0]);
+    }
+    //idgraph_identities[idgraph_identities.length - 1] is prime identity,don't need to compare
+    assert.isTrue(
+        isArrayEqual(event_identities, idgraph_identities.slice(0, idgraph_identities.length - 1)),
+        'event identities should be equal to idgraph identities'
+    );
+
+    const data = eventDatas[eventDatas.length - 1];
+    for (let i = 0; i < eventDatas[eventDatas.length - 1].idGraph.length; i++) {
+        if (isEqual(data.idGraph[i][0], data.identity)) {
+            assert.isTrue(data.idGraph[i][1].is_verified, 'identity should be verified');
+        }
+    }
+    assert.equal(data?.who, u8aToHex(signer.addressRaw), 'check caller error');
+}
+
+export function assertIdentityCreated(signer: KeyringPair, identityEvent: IdentityGenericEvent | undefined) {
+    assert.equal(identityEvent?.who, u8aToHex(signer.addressRaw), 'check caller error');
+}
+
+export function assertIdentityRemoved(signer: KeyringPair, identityEvent: IdentityGenericEvent | undefined) {
+    assert.equal(identityEvent?.idGraph, null, 'check idGraph error,should be null after removed');
+    assert.equal(identityEvent?.who, u8aToHex(signer.addressRaw), 'check caller error');
+}
+
+export async function checkErrorDetail(
+    response: string[] | Event[],
+    expectedDetail: string,
+    isModule: boolean
+): Promise<boolean> {
+    let detail: string = '';
+    // TODO: sometimes `item.data.detail.toHuman()` or `item` is treated as object (why?)
+    //       I have to JSON.stringify it to assign it to a string
+    response.map((item: any) => {
+        isModule ? (detail = JSON.stringify(item.data.detail.toHuman())) : (detail = JSON.stringify(item));
+        assert.isTrue(
+            detail.includes(expectedDetail),
+            `check error detail failed, expected detail is ${expectedDetail}, but got ${detail}`
+        );
+    });
+    return true;
+}
+
+export async function buildValidations(
```

Comment:
I feel that it would be better to move this to `identity.ts`. This is the functionality for constructing identity validations.